### PR TITLE
gRPC -  align service names generated by quarkus with grpc-java

### LINKED ISSUE #27170: Generated grpc code cannot be compiled if a underscore is in the name of a method
### Describe the bug

I got a proto file which contains a rpc method with a underscore in the name

```
syntax = "proto3";

package mypackage;

option java_multiple_files = true;
option java_package = "mypackage";

service MyService {
  // Get info about the service
  rpc service_info(ServiceInfoRequest) returns (ServiceInfo) {}
}
message ServiceInfo {
  string software_version = 1;
}
message ServiceInfoRequest {
}
```

But the generated code is very inconsistend and cannot be compiled

e.g. in `MyServiceImplBase implements io.grpc.BindableService` its `getServiceinfoMethod` instead of `getServiceInfoMethod`

```
        @java.lang.Override
        public io.grpc.ServerServiceDefinition bindService() {
            return io.grpc.ServerServiceDefinition.builder(getServiceDescriptor()).addMethod(mypackage.MyServiceGrpc.getServiceinfoMethod(), asyncUnaryCall(new MethodHandlers<mypackage.ServiceInfoRequest, mypackage.ServiceInfo>(this, METHODID_SERVICE_INFO, compression))).build();
        }
```

causes

```
[ERROR]   generated-sources/grpc/mypackage/MutinyMyServiceGrpc.java:[73,117] cannot find symbol
[ERROR]   symbol:   method getServiceinfoMethod()
[ERROR]   location: class mypackage.MyServiceGrpc
```

and in ``MutinyMyServiceStub extends io.grpc.stub.AbstractStub<MutinyMyServiceStub> implements io.quarkus.grpc.MutinyStub`
it is `delegateStub::service_info` instead of `delegateStub::serviceInfo`

```
        public io.smallrye.mutiny.Uni<mypackage.ServiceInfo> service_info(mypackage.ServiceInfoRequest request) {
            return io.quarkus.grpc.stubs.ClientCalls.oneToOne(request, delegateStub::service_info);
        }
```

causes
```
[ERROR]   generated-sources/grpc/mypackage/MutinyMyServiceGrpc.java:[44,72] invalid method reference
[ERROR]   cannot find symbol
[ERROR]     symbol:   method service_info()
[ERROR]     location: class mypackage.MyServiceGrpc.MyServiceStub
```

### Expected behavior

Generated code is consistent and can be compiled

### Actual behavior

_No response_

### How to Reproduce?

Download Quarkus sample for grpc.

Add proto example file with name my-service.proto into /src/main/proto, see an example above.

Run mvn package

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #27170

### CODE REVIEW

File: extensions/grpc/protoc/src/main/java/io/quarkus/grpc/protoc/plugin/MutinyGrpcGenerator.java [hunk 1]
```diff
@@ -171,8 +174,31 @@ private MethodContext buildMethodContext(DescriptorProtos.MethodDescriptorProto
...
-    private String lowerCaseFirst(String s) {
-        return Character.toLowerCase(s.charAt(0)) + s.substring(1);
+    static String adaptMethodName(String name) {
+        // We need to adjust the method name in the same way as the grpc-java compiler does:
+        // 1. Decapitalize
+        // 2. Replace an underscore and capitalize the following letter
+        // 3. Append the underscore for a java keyword
+        // (these rules are bit odd but we need to follow them unless we get rid of the grpc-java compiler)
+        // https://github.com/quarkusio/quarkus/issues/27170
+        name = Introspector.decapitalize(name);
```
Comment (mkouba) [inline extensions/grpc/protoc/src/main/java/io/quarkus/grpc/protoc/plugin/MutinyGrpcGenerator.java:184]: I'm not quite sure whether it's legal (from JPMS point of view) to use `java.beans.Introspector` and `javax.lang.model.SourceVersion` in this method...

Comment (mkouba) [reply extensions/grpc/protoc/src/main/java/io/quarkus/grpc/protoc/plugin/MutinyGrpcGenerator.java:184]: The thing is that this code should only be used during the build when the grpc stubs are generated.

Comment (mkouba) [reply extensions/grpc/protoc/src/main/java/io/quarkus/grpc/protoc/plugin/MutinyGrpcGenerator.java:184]: But can easily copy the code in our own utils if necessary...

Comment (mkouba) [reply extensions/grpc/protoc/src/main/java/io/quarkus/grpc/protoc/plugin/MutinyGrpcGenerator.java]: Done.

**Review by michalszynkiewicz — APPROVED** [review]

**Pull request merged**